datalab_cohorts/__init__.py

```diff
@@ -1637,19 +1586,48 @@ def with_death_recorded_in_cpns(
         between=None,
         # Set return type
         returning="binary_flag",
-        # If we're returning a date, how granular should it be?
+        date_format=None,
+        # Deprecated options kept for now for backwards compatibility
         include_month=False,
         include_day=False,
     ):
-        validate_time_period_options(**locals())
-        return "with_death_recorded_in_cpns", locals()
+        return "with_death_recorded_in_cpns", process_arguments(locals())
 
     @staticmethod
     def date_of(
-        source, include_month=False, include_day=False, return_expectations=None
+        source, return_expectations=None,
+        date_format=None,
+        # Deprecated options kept for now for backwards compatibility
+        include_month=False,
+        include_day=False,
     ):
         returning = "date"
-        return "value_from", locals()
+        return "value_from", process_arguments(locals())
+
+
+def process_arguments(args):
```

Comment:
A docstring would be useful here as a hint to future coders about what the expected responsibilities of this function are